packages/common/src/codegen/utils/contractToInterface.ts

```diff
@@ -39,13 +49,17 @@ export function contractToInterface(
   symbolImports: SymbolImport[];
   qualifiedSymbols: Map<string, QualifiedSymbol>;
 } {
-  let ast: SourceUnit;
-  try {
-    ast = parse(source);
-  } catch (error) {
-    throw new MUDError(`Failed to parse contract ${contractName}: ${error}`);
+  const parser = Parser.create("0.8.24");
```

Comment:
Is there a way to avoid hardcoding the version here?

Reply:
I believe we already parse the foundry config and pass that through in a few places in MUD tools, so we could pass that through here to avoid hardcoding it

Reply:
I wasn't sure what the version policy wrt Solidity was for MUD. Is it bound to the MUD release, or is completely user defined? Otherwise, Slang can [infer the version](https://nomicfoundation.github.io/slang/latest/user-guide/04-getting-started/02-choosing-a-solidity-version/#inferring-compatible-solidity-versions) from the source code, or we can extract it as a constant, or pass it through as a parameter. Happy to implement whatever solution works best for you.

Reply:
@alvrs @frolic  I noticed that the currently used solc version in this repository is updated manually every now and then. Example: https://github.com/latticexyz/mud/commit/aabd30767cdda7ce0c32663e7cc483db1b66d967

In that commit, while projects/existing code specifically use `0.8.24`, the Solidity files contain a more permissive pragma `pragma solidity >=0.8.24;`. When using the Slang parser, the version can be hard-coded/updated manually along the rest of the codebase like the commit above, or inferred automatically (as suggested by @ggiraldez's comment here). But with inference, a higher version could be inferred, since it would match `>=0.8.24`. Up to you on which solution you prefer.

Please let us know if you have any other feedback.